src/main/java/eu/dissco/core/digitalspecimenprocessor/service/HandleService.java

```diff
@@ -30,62 +64,261 @@
+                  + "Local identifier already has persistent identifier: %s",
+              digitalSpecimen.physicalSpecimenId(),
+              new String(optionalRecord.get().get(HANDLES.HANDLE), StandardCharsets.UTF_8)));
+    }
+  }
+
+  private List<HandleAttribute> fillFdoRecord(DigitalSpecimen digitalSpecimen, String handle,
+      Instant recordTimeStamp)
       throws TransformerException {
-    var handleAttributes = new ArrayList<HandleAttribute>();
-    handleAttributes.add(
-        new HandleAttribute(1, "pid",
-            ("https://hdl.handle.net/" + handle).getBytes(StandardCharsets.UTF_8)));
-    handleAttributes.add(new HandleAttribute(2, "pidIssuer",
-        createPidReference("https://doi.org/10.22/10.22/2AA-GAA-E29", "DOI", "RA Issuing DOI")));
-    handleAttributes.add(new HandleAttribute(3, "digitalObjectType",
-        createPidReference(DUMMY_HANDLE, HANDLE, "Digital Specimen")));
-    handleAttributes.add(new HandleAttribute(4, DIGITAL_OBJECT_SUBTYPE,
-        createPidReference("https://hdl.handle.net/21...", HANDLE, digitalSpecimen.type())));
-    handleAttributes.add(new HandleAttribute(5, "10320/loc", createLocations(handle)));
-    handleAttributes.add(new HandleAttribute(6, "issueDate", createIssueDate(recordTimestamp)));
-    handleAttributes.add(
-        new HandleAttribute(7, "issueNumber", "1".getBytes(StandardCharsets.UTF_8)));
-    handleAttributes.add(
-        new HandleAttribute(8, "pidStatus", "DRAFT".getBytes(StandardCharsets.UTF_8)));
-    handleAttributes.add(
-        new HandleAttribute(11, "pidKernelMetadataLicense",
-            "https://creativecommons.org/publicdomain/zero/1.0/".getBytes(StandardCharsets.UTF_8)));
-    handleAttributes.add(new HandleAttribute(14, "digitalOrPhysical", "physical".getBytes(
-        StandardCharsets.UTF_8)));
-    handleAttributes.add(new HandleAttribute(15, SPECIMEN_HOST, createPidReference(
-        digitalSpecimen.attributes().get("ods:organisationId").asText(), "ROR", TO_BE_FIXED)));
-    handleAttributes.add(new HandleAttribute(100, "HS_ADMIN", decodeAdmin()));
-    return handleAttributes;
-  }
-
-  private byte[] createIssueDate(Instant recordTimestamp) {
-    SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd");
-    return formatter.format(Date.from(recordTimestamp)).getBytes(StandardCharsets.UTF_8);
+    List<HandleAttribute> fdoRecord = new ArrayList<>();
+    fdoRecord.addAll(fillFdoRecordGeneratedAttributes(handle, recordTimeStamp));
+    fdoRecord.addAll(fillFdoRecordSpecimenAttributes(digitalSpecimen));
+    return fdoRecord;
+  }
+
+  private List<HandleAttribute> fillFdoRecordGeneratedAttributes(String handle,
+      Instant recordTimeStamp)
+      throws TransformerException {
+    List<HandleAttribute> fdoRecord = new ArrayList<>();
+
+    // 100: Admin Handle
+    fdoRecord.add(
+        new HandleAttribute(FIELD_IDX.get(HS_ADMIN), HS_ADMIN, decodeAdmin()));
+
+    // 101: 10320/loc
+    byte[] loc = createLocations(handle);
```

Comment:
We should maybe add both the API as the sandbox. Think sandbox first right? Would be nice to add them in this PR, otherwise we can add it later

Reply:
updated createLocations